tensorlayer/db.py

```diff
@@ -5,11 +5,725 @@
+        """Deletes validation log.
+
+        Parameters
+        -----------
+        kwargs : logging information
+            Find items to delete, leave it empty to delete all log.
+
+        Examples
+        ---------
+        - see ``train_log``.
+        """
+        self._fill_project_info(kwargs)
+        self.db.ValidLog.delete_many(kwargs)
+        logging.info("[Database] Delete ValidLog SUCCESS")
+
+    def del_test_log(self, **kwargs):
+        """Deletes testing log.
+
+        Parameters
+        -----------
+        kwargs : logging information
+            Find items to delete, leave it empty to delete all log.
+
+        Examples
+        ---------
+        - see ``train_log``.
+        """
+        self._fill_project_info(kwargs)
+        self.db.TestLog.delete_many(kwargs)
+        logging.info("[Database] Delete TestLog SUCCESS")
+
+    ## =========================== JOB =================================== ##
+    def push_task(self, task_key=None, script=None, hyper_parameters=None, result_key=None, **kwargs):
+        """Uploads a task to the database, timestamp will be added automatically.
+
+        Parameters
+        -----------
+        task_key : str
+            The task name.
+        script : str
+            File name of the python script.
+        hyper_parameters : dictionary
+            The hyper parameters pass into the script.
+        kwargs : other parameters
+            Users customized parameters such as description, version number.
+
+        Examples
+        -----------
+        - Uploads a task
+        >>> db.push_task(task_key='mnist', script='example/tutorial_mnist_simple.py', description='simple tutorial')
+
+        - Finds and runs the latest task
+        >>> db.run_one_task(sess=sess, sort=[("time", pymongo.DESCENDING)])
+        >>> db.run_one_task(sess=sess, sort=[("time", -1)])
+
+        - Finds and runs the oldest task
+        >>> db.run_one_task(sess=sess, sort=[("time", pymongo.ASCENDING)])
+        >>> db.run_one_task(sess=sess, sort=[("time", 1)])
+
+        """
+        if not isinstance(task_key, str):  # is None:
+            raise Exception("task_key should be string")
+        if not isinstance(script, str):  # is None:
+            raise Exception("script should be string")
+        if hyper_parameters is None:
+            hyper_parameters = {}
+        if result_key is None:
+            result_key = []
+
+        self._fill_project_info(kwargs)
+        kwargs.update({'time': datetime.utcnow()})
+        kwargs.update({'hyper_parameters': hyper_parameters})
+        kwargs.update({'result_key': result_key})
+
+        _script = open(script, 'rb').read()
+
+        kwargs.update({'status': 'pending', 'script': _script, 'result': {}})
+        self.db.Task.insert_one(kwargs)
+        logging.info("[Database] Saved Task - task_key: {} script: {}".format(task_key, script))
+
+    def run_one_task(self, task_key=None, sort=None, **kwargs):
```

Comment:
run_task

Reply:
this one only run one task, i think run_one_task is better?